neptune/new/internal/backends/hosted_client.py

```diff
@@ -0,0 +1,175 @@
+    }
+}
+
+
+def create_http_client(ssl_verify: bool, proxies: Dict[str, str]) -> RequestsClient:
+    http_client = RequestsClient(ssl_verify=ssl_verify)
+    http_client.session.verify = ssl_verify
+
+    update_session_proxies(http_client.session, proxies)
+
+    user_agent = 'neptune-client/{lib_version} ({system}, python {python_version})'.format(
+        lib_version=neptune_client_version,
+        system=platform.platform(),
+        python_version=platform.python_version())
+    http_client.session.headers.update({'User-Agent': user_agent})
+
+    return http_client
+
+
+@cache
+def _get_token_client(
+        credentials: Credentials,
+        ssl_verify: bool,
+        proxies: Dict[str, str],
+        endpoint_url: str = None
+) -> SwaggerClient:
+    config_api_url = credentials.api_url_opt or credentials.token_origin_address
+    if proxies is None and endpoint_url is None:
+        verify_host_resolution(config_api_url)
+
+    token_http_client = create_http_client(ssl_verify, proxies)
+
+    return create_swagger_client(
+        build_operation_url(endpoint_url or config_api_url, BACKEND_SWAGGER_PATH),
+        token_http_client
+    )
+
+
+@cache
+@with_api_exceptions_handler
+def get_client_config(credentials: Credentials, ssl_verify: bool, proxies: Dict[str, str]) -> ClientConfig:
+    backend_client = _get_token_client(credentials=credentials, ssl_verify=ssl_verify, proxies=proxies)
+
+    config = backend_client.api.getClientConfig(
+        X_Neptune_Api_Token=credentials.api_token,
+        alpha="true",
+        **DEFAULT_REQUEST_KWARGS,
+    ).response().result
+
+    if hasattr(config, "pyLibVersions"):
+        min_recommended = getattr(config.pyLibVersions, "minRecommendedVersion", None)
+        min_compatible = getattr(config.pyLibVersions, "minCompatibleVersion", None)
+        max_compatible = getattr(config.pyLibVersions, "maxCompatibleVersion", None)
+    else:
+        raise UnsupportedClientVersion(neptune_client_version, max_version="0.4.111")
+
+    return ClientConfig(
+        api_url=config.apiUrl,
+        display_url=config.applicationUrl,
+        min_recommended_version=version.parse(min_recommended) if min_recommended else None,
+        min_compatible_version=version.parse(min_compatible) if min_compatible else None,
+        max_compatible_version=version.parse(max_compatible) if max_compatible else None
+    )
+
+
+@cache
+def create_http_client_with_auth(
+        credentials: Credentials,
+        ssl_verify: bool,
+        proxies: Dict[str, str]
+) -> Tuple[RequestsClient, ClientConfig]:
+    client_config = get_client_config(
+        credentials=credentials,
+        ssl_verify=ssl_verify,
+        proxies=proxies
+    )
+    verify_client_version(client_config, neptune_client_version)
+
+    config_api_url = credentials.api_url_opt or credentials.token_origin_address
+    if proxies is None:
+        verify_host_resolution(config_api_url)
```

Comment:
Is it necessary here? I think it's already verified inside get_client_config.

Reply:
Sure